src/hoc/withAuthRedirect.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+const mapStateToPropsForRedirect = (state) => ({
+  isAuth: state.auth.isAuth,
+});
+
+const withAuthRedirect = (Component) => {
+  class RedirectComponent extends React.Component {
+    render() {
+      const { props } = this;
+      if (!props.isAuth) return <Redirect to="/login" />;
+      return <Component {...props} />;
+    }
+  }
+
+  const ConnectedAuthRedirectComponent = connect(mapStateToPropsForRedirect)(RedirectComponent);
```

Comment:
Consider using the "return connect(...)..." approach for the sake of consistency.

Reply:
/fxd